hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnectionManager.java

```diff
@@ -193,25 +202,77 @@ private synchronized boolean bind(TcpIpConnection connection, Address remoteEndP
         if (logger.isFinestEnabled()) {
             logger.finest("Binding " + connection + " to " + remoteEndPoint + ", reply is " + reply);
         }
+
         final Address thisAddress = ioService.getThisAddress();
-        if (ioService.isSocketBindAny() && !connection.isClient() && !thisAddress.equals(localEndpoint)) {
-            String msg = "Wrong bind request from " + remoteEndPoint
-                    + "! This node is not requested endpoint: " + localEndpoint;
-            logger.warning(msg);
-            connection.close(msg, null);
+
+        // Some simple spoofing attack prevention
+        // Prevent BINDs from src that doesn't match the BIND local address
+        // Prevent BINDs from src that match us (same host & port)
+        if (SPOOFING_CHECKS
+                && (ensureValidBindSource(connection, remoteEndPoint)
+                 || ensureBindNotFromSelf(connection, remoteEndPoint, thisAddress))) {
+            return false;
+        }
+
+        // Prevent BINDs that don't have this node as the destination
+        if (ensureValidBindTarget(connection, remoteEndPoint, localEndpoint, thisAddress)) {
```

Comment:
just a minor comment:   I think `ensure*()` methods should return `false` when parameters are not valid. I mean return values should be reverse.